### DISCUSSION TITLE
Unwanted recomposition on lazy row scroll

### DISCUSSION THREAD

**Discussion by thedipsy** [coil-kt/coil#1991] [Q&A]
Hi, I render images in a lazy row and on each scroll back and forth, the images are partially recomposing, causing flickering with loading state to some items state even though the image was previously loaded.

I would like to avoid the recomposition.

This is part of the code for the asyncImagePainter

```
  var retryHash by remember { mutableIntStateOf(0) }
  val imagePainter = rememberAsyncImagePainter(
    ImageRequest.Builder(LocalContext.current)
      .data(data = url)
      .memoryCacheKey(url)
      .memoryCachePolicy(CachePolicy.ENABLED)
      .setParameter(KEY_RETRY_HASH, retryHash, null)
      .build()
  )

```

How can I fix this ?

**Comment by mainrs** [comment]
I have the same issue. However, mine occurs if the parent component/window is resized and causes a recomposition of the image. The image flickers for a moment with the placeholder until the real, cached image is provided.
